specs/_features/eip7594/polynomial-commitments-sampling.md

```diff
@@ -180,6 +179,31 @@ def fft_field(vals: Sequence[BLSFieldElement],
         return _fft_field(vals, roots_of_unity)
 ```
 
+#### `coset_fft_field`
+
+```python
+def coset_fft_field(vals: Sequence[BLSFieldElement], 
+                    roots_of_unity: Sequence[BLSFieldElement], 
+                    inv: bool=False) -> Sequence[BLSFieldElement]:
+    
+    # Multiply each entry in `vals` by succeeding powers of `factor`
+    # ie [vals[0] * factor^0, vals[1] * factor^1, ..., vals[n] * factor^n]
+    def shift_vals(vals: Sequence[BLSFieldElement], factor: BLSFieldElement) -> Sequence[BLSFieldElement]:
+        shift = 1
```

Comment:
I'm not opposed to it, but I don't think we typically use nested functions in the specs.